src/CMakeLists.txt

```diff
@@ -1,549 +1,534 @@
-set(CMAKE_CXX_FLAGS "${Qt5XmlPatterns_EXECUTABLE_COMPILE_FLAGS}")
-set(CMAKE_CXX_FLAGS "${Qt5PrintSupport_EXECUTABLE_COMPILE_FLAGS}")
-set(CMAKE_CXX_FLAGS "${Qt5WebSockets_EXECUTABLE_COMPILE_FLAGS}")
-set(CMAKE_CXX_FLAGS "${Qt5Qml_EXECUTABLE_COMPILE_FLAGS}")
-set(CMAKE_CXX_FLAGS "${Qt5X11Extras_EXECUTABLE_COMPILE_FLAGS}")
-set(CMAKE_CXX_FLAGS "-std=c++0x")
-
-# The Qt5Widgets_LIBRARIES variable also includes QtGui and QtCore
-target_link_libraries(QOwnNotes ${Qt5Widgets_LIBRARIES} ${Qt5Sql_LIBRARIES}
-        ${Qt5Svg_LIBRARIES} ${Qt5Network_LIBRARIES}
-        ${Qt5Xml_LIBRARIES} ${Qt5XmlPatterns_LIBRARIES}
-        ${Qt5PrintSupport_LIBRARIES} ${Qt5WebSockets_LIBRARIES}
-        ${Qt5Qml_LIBRARIES} ${Qt5X11Extras_LIBRARIES} ${X11_LIBRARIES})
-
-#qt5_use_modules(QOwnNotes Widgets Core Gui Sql Svg Network Script Xml PrintSupport)
-
-# Sonnet support if we can get it to run in the future
-#find_package(KF5Sonnet)
-#target_link_libraries(KF5::SonnetCore KF5::SonnetUi)
-
-add_compile_definitions(QAPPLICATION_CLASS=QApplication)
+cmake_minimum_required(VERSION 3.2.2)
+project(QOwnNotes)
+
+# include some Botan settings
+add_subdirectory(libraries/botan)
+#include sonnet core directory, required for building plugins
+include_directories(libraries/sonnet/src/core)
+#some hunspell settings for windows
+include(libraries/sonnet/src/plugins/hunspell/hunspell/CMakeLists.txt)
+
+
+add_executable(QOwnNotes "")
+
+set_target_properties(QOwnNotes PROPERTIES
+    CXX_STANDARD 11
+    CXX_STANDARD_REQUIRED ON
+    AUTOMOC ON
+    AUTOUIC ON
+    AUTORCC ON
+)
+
+target_include_directories(QOwnNotes PRIVATE
+    ${CMAKE_CURRENT_LIST_DIR}
+    ${CMAKE_CURRENT_BINARY_DIR}
+)
+
+# Disable shared library building of libraries
+set(BUILD_SHARED_LIBS OFF)
+# QHotKey library
+# Disable building tests inside QHotKey library
+set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
+set(QHOTKEY_EXAMPLES OFF)
+add_subdirectory(libraries/qhotkey)
+
+# FakeVim library
+set(CREATE_STATIC_LIBRARY ON)
+if(MSVC)
+    add_compile_definitions(FAKEVIM_STATIC_DEFINE QTCREATOR_UTILS_STATIC_LIB)
+endif()
+add_subdirectory(libraries/fakevim)
+
+
+find_package(Qt5 COMPONENTS REQUIRED
+    LinguistTools
+    Core
+    Widgets
+    Gui
+    Sql
+    Svg
+    Network
+    Xml
+    XmlPatterns
+    PrintSupport
+    WebSockets
+    Qml
+)
+if(UNIX)
+    find_package(Qt5 COMPONENTS REQUIRED X11Extras)
+    find_package( X11 REQUIRED )
+endif()
+
+find_program(CCACHE_FOUND ccache)
+
+if(CCACHE_FOUND)
+    message(STATUS "ccache found")
+    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
+    set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
+endif(CCACHE_FOUND)
+
+set(RESOURCE_FILES
+    breeze-qownnotes.qrc
+    breeze-dark-qownnotes.qrc
+    qownnotes.qrc
+    demonotes.qrc
+    images.qrc
+    texts.qrc
+    configurations.qrc
+    libraries/qmarkdowntextedit/media.qrc
+    libraries/qdarkstyle/style.qrc
+    libraries/qkeysequencewidget/qkeysequencewidget/qkeysequencewidget.qrc

```

Comment:
can we not do this? `qkeysequencewidget` has a cmake file already, we can just use the variables from there or link to it instead

Reply:
I could not do all changes in one evening.